### ISSUE TITLE
Dot Products and Vector (Cross) Products Are Not Distinguished in Speech

### ISSUE THREAD

**Issue by NV-Codes** [NSoiffer/MathCAT#323]
## MathCAT Version

MathCAT-0.6.6 (NVDA Add-On)

## Description of Issue

MathCAT's speech verbalizes both the dot multiplication symbol and the cross multiplication symbol as "times," as well as some implicit multiplication (multiplication implied by adjacency). An issue arises in calculus and physics when the dot and cross symbols mean different types of vector operations, and failing to disambiguate the type of multiplication greatly hinders proper understanding of math content.

For instance, the dot product of vector A and vector B ($\vec{A} \cdot \vec{B}$) and the cross product between vector A and vector B ($\vec{A} \times \vec{B}$) are both verbalized as "vector cap A times vector cap B."

## Proposed Solution

A checkbox in MathCAT's "Speech" settings could allow the user to toggle disambiguation of dot and cross multiplication.

+ [x] Verbalize dot and cross multiplication differently
+ [ ] Verbalize dot and cross multiplication differently

Simply verbalizing "dot" and "cross" for the respective symbols when the option is checked should suffice, as that seems clear and concise.

**Comment by NSoiffer** [comment]
Thanks for your suggestion. I'm not keen on adding lots of "little" tweaks in the UI. The more that get added, the harder it becomes to find any individual feature. Also, most people never change from the defaults.

What might work is trying to detect that the operands are vectors. The check could be that the operands are either bold letters or letters with a right arrow or harpoon above. In those cases say either "dot" or "cross". Would that solve this issue for you?

**Comment by NV-Codes** [comment]
Yes, I think that would solve the issue, as long as letters with hats are included as well. For example, $\hat{x} \times \hat{y} = \hat{z}$ is a relation of unit vectors in a right-handed Cartesian coordinate system.

**Comment by NV-Codes** [comment]
It would also be important to consider formulas like the following:

1. Magnetic Flux through a Surface: $\Phi_B = \oint \mathbf B \cdot d \mathbf A$
2. Magnetic Field Produced by a Current-Carrying Loop ($C_2$): $\mathbf B_2 (\mathbf r_1) = \frac{\mu_0}{4\pi} \oint \limits_{C_2} I_2 d\mathbf s_2 \times \frac{\mathbf r_1 - \mathbf r_2}{|\mathbf r_1 - \mathbf r_2|^3}$
3. Force on Loop 1 ($C_1$) by Loop 2 ($C_2$): $\mathbf F_1 = -\frac{\mu_0}{4\pi} \oint \limits_{C_1} I_1 d \mathbf s_1 \cdot \oint \limits_{C_2} I_2 d \mathbf s_2 \frac{\mathbf r_1 - \mathbf r_2}{|\mathbf r_1 - \mathbf r_2|^3}$

The operands could be differential vectors, vector fractions, or even integrals that represent vector quantities.

**Comment by NSoiffer** [comment]
Thanks for the extra examples, I would have implemented something that is too restrictive. It seems if the operands _contain_ (versus _are_) a bold letter or have a right arrow/harpoon over a letter, then use "dot" or "cross". I think _both_ operands need to contain a vector. Is that correct, or just one of them?

Note: MathCAT parses the expression so that integrals and their integrands are operands of the dot product/cross products above. Similarly, $d \mathbf A$ is an operand.

**Comment by NV-Codes** [comment]
Yes, both operands must be vectors, and checking for containment of vector identifiers should work! A note is that the pseudovector $\nabla$ can also be an operand; though "gradient of," "divergence of," and "curl of" should cover expressions with $\nabla$ in the vast majority of cases, there are times when $\nabla'$ or $\nabla_r$ are used to indicate that the vector differential is acting with respect to a particular set of coordinates (e.g., $x'$, $y'$, and $z'$ instead of $x$, $y$, and $z$).

**Comment by NV-Codes** [comment]
I think it also makes sense for MathCAT to verbalize "dot" and "cross" in character-by-character navigation (or whenever the symbol is in focus during navigation).

**Comment by NSoiffer** [comment]
In character mode, I pretty strongly feel that semantic interpretation is a bad idea. That leaves open the issue of what to call these without interpreting these. I will open a separate issue for that.
